src/main/groovy/io/openliberty/tools/gradle/tasks/AbstractServerTask.groovy

```diff
@@ -745,57 +742,85 @@ abstract class AbstractServerTask extends AbstractTask {
         Properties configuredProps = combineServerEnvProperties(server.env, envProjectProps);
 
         if(server.appendServerEnv) {
-            setServerEnvWithAppendServerEnvHelper(envFile, serverEnvPath, configuredProps)
+            return setServerEnvWithAppendServerEnvHelper(envFile, serverEnvPath, configuredProps)
         }
         else {
-            setServerEnvHelper(envFile, serverEnvPath, configuredProps)
+            return setServerEnvHelper(envFile, serverEnvPath, configuredProps)
         }
     }
 
-    private void setServerEnvWithAppendServerEnvHelper(File envFile, String serverEnvPath, Properties configuredProps) {
+    private String setServerEnvWithAppendServerEnvHelper(File envFile, String serverEnvPath, Properties configuredProps) {
         Properties serverEnvProps = convertServerEnvToProperties(envFile);
         Properties mergedProperties = new Properties();
 
         if (server.serverEnvFile != null && server.serverEnvFile.exists()) {
             if (serverEnvPath != null) {
-                logger.warn("The " + serverEnvPath + " file is merged with the " + server.serverEnvFile.getCanonicalPath() + " file.")
+                logger.debug("The serverEnvFile "+ server.serverEnvFile.getCanonicalPath() + " is merged with the " + serverEnvPath + " file.")
+            }
+            else {
+                logger.debug("The serverEnvFile "+ server.serverEnvFile.getCanonicalPath() + " is merged with the " + getServerDir(project).getCanonicalPath() + " file.")
             }
             Properties configuredServerEnvProps = convertServerEnvToProperties(server.serverEnvFile);
             //Merge with either default server.env or with what has already been merged if
             mergedProperties = (Properties) combineServerEnvProperties(serverEnvProps, configuredServerEnvProps);
         }
 
-        if ((server.env != null && !server.env.isEmpty()) || !envProjectProps.isEmpty()) {
+        if (!configuredProps.isEmpty()) {
             if (serverEnvPath != null) {
-                logger.warn("The " + serverEnvPath + " file is merged with inlined configuration.")
+                logger.debug("The " + serverEnvPath + " file is merged with inlined configuration.")
+            }
+            else {
+                logger.debug("The " + getServerDir(project).getCanonicalPath() + " file is merged with inlined configuration.")
             }
 
             mergedProperties = combineServerEnvProperties(mergedProperties, configuredProps);
         }
 
         if(!mergedProperties.isEmpty()) {
             writeServerEnvProperties(envFile, mergedProperties);
+            return setServerEnvPathHelperForAppendServerEnv(envFile, configuredProps, serverEnvPath)
+        }
+
+        return serverEnvPath;
+    }
+
+    private String setServerEnvPathHelperForAppendServerEnv(File envFile, Properties configuredProps, String serverEnvPath) {
+        boolean configDirEnvMerged = envFile != null;
```

Comment:
envFile is the server.env in the server directory, which may or may not have had a server.env from the configDirectory merged with it.